parl/algorithms/torch/oac.py

```diff
@@ -0,0 +1,174 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import parl
+import torch
+import math
+from torch.distributions import Normal
+import torch.nn.functional as F
+from copy import deepcopy
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+__all__ = ['OAC']
+
+
+class OAC(parl.Algorithm):
+    def __init__(self,
+                 model,
+                 max_action,
+                 discount=None,
+                 tau=None,
+                 alpha=None,
+                 beta=None,
+                 delta=None,
+                 actor_lr=None,
+                 critic_lr=None,
+                 decay=None,
+                 policy_freq=1,
+                 automatic_entropy_tuning=False,
+                 entropy_lr=None,
+                 action_space=None):
+        assert isinstance(discount, float)
+        assert isinstance(tau, float)
+        assert isinstance(alpha, float)
+        assert isinstance(actor_lr, float)
+        assert isinstance(critic_lr, float)
+        assert isinstance(entropy_lr, float)
```

Comment:
Is `entropy_lr` a necessary parameter or not ? if not, please remove this assert line.

Please check all the necessary parameter, and do the same check in SAC.

Reply:
done